### PR TITLE
OVMM: Merge episode generation updates from main

### PR DESCRIPTION
**Pull request by JHurricane96**
## Motivation and Context

Merging following PRs:

#1108, #1248, #1249, #1298, #1301

## Types of changes

- **\[Docs change\]** Addition or changes to the documentation
- **\[Refactoring\]** Large changes to the code that improve its functionality or performance
- **\[Dependency Upgrade\]** Upgrades one or several dependencies in habitat
- **\[Bug Fix\]** (non-breaking change which fixes an issue)
- **\[Development\]** A pull request that add new features to the [habitat-lab](/habitat-lab) task and environment codebase. Development Pull Requests must be small (less that 500 lines of code change), have unit testing, very extensive documentation and examples. These are typically new tasks, environments, sensors, etc... The review process for these Pull Request is longer because these changes will be maintained by our core team of developers, so make sure your changes are easy to understand!
- **\[Experiment\]** A pull request that add new features to the [habitat-baselines](/habitat-baselines/) training codebase. Experiments Pull Requests can be any size, must have smoke/integration tests and be isolated from the rest of the code. Your code additions should not rely on other habitat-baselines code. This is to avoid dependencies between different parts of habitat-baselines. You must also include a README that will document how to use your new feature or trainer. **You** will be the maintainer of this code, if the code becomes stale or is not supported often enough, we will eventually remove it.

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes if required.

### CODE REVIEW

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py [hunk 1]
```diff
@@ -430,83 +435,91 @@ def add_cat_fields_to_episodes(
...
+                sim,
+                episode,
+                obj_idx_to_name,
+                rec_viewpoints,
+                rec_to_parent_obj,
+                debug_viz,
+            ):
+                print("Skipping episode", episode["episode_id"])
+                continue
+            assert (
```
Comment (Skylion007) [inline habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:514]: Can be split into separate assertions.

File: habitat-lab/habitat/datasets/rearrange/rearrange_generator.py [hunk 2]
```diff
@@ -614,6 +647,15 @@ def generate_single_episode(self) -> Optional[RearrangeEpisode]:
             if len(object_sample_data) == 0:
                 return None
             new_objects, receptacles = zip(*object_sample_data)
+            # collect names of all newly placed target objects
+            target_object_names.extend(
+                [
```
Comment (Skylion007) [inline habitat-lab/habitat/datasets/rearrange/rearrange_generator.py:652]: Should be generator

**Comment by Skylion007** [comment]
Ah, sorry saw thi is a merge PR, ignore my comments.

**Review by Skylion007 — APPROVED** [review]

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py [hunk 3]
```diff
@@ -1,12 +1,17 @@
 import argparse
+import ctypes
```
Comment (ykarmesh) [inline habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:2]: Didnt find the changes for this file in the 5 PRs that you have tagged

Comment (ykarmesh) [reply habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:2]: Nevermind I think I got it.

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py [hunk 4]
```diff
@@ -430,83 +435,91 @@ def add_cat_fields_to_episodes(
...
-        )
-        obj_cat, start_rec_cat, goal_rec_cat = get_obj_rec_cat_in_eps(
-            episode,
-            obj_category_mapping=obj_category_mapping,
-            rec_category_mapping=rec_category_mapping,
-        )
+    with tqdm(total=num_episodes) as pbar:
+        for episode in episodes["episodes"]:
+            scene_id = episode["scene_id"]
+            episode["additional_obj_config_paths"] = ['data/objects/amazon_berkeley/configs/', 'data/objects/google_scanned/configs/', 'data/objects/ai2thorhab/configs/objects', 'data/objects/floorplanner/configs/objects']
```
Comment (ykarmesh) [inline habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:441]: Seeing this reminds me to ask you: Are you using YCB in the val set?

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py (hunk 3 above)
Comment (JHurricane96) [reply habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py:2]: There are some additional changes I had to make on top of those PRs to get things working on our branch. You can look at the commits that were made exclusively by me for these changes, and the ones that were part of the 5 PRs were made by Alex.

File: habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py (hunk 4 above)
Comment (JHurricane96) [reply habitat-lab/habitat/datasets/rearrange/modify_episodes_for_object_rearrange.py]: I'm not. Thanks for adding a comment here, this code shouldn't be here. I've removed it now.

File: test/test_rearrange_task.py [hunk 5]
```diff
@@ -439,46 +245,155 @@ def test_receptacle_parsing(debug_visualization):
...
-            for _samp_ix in range(num_samples):
-                rec_samples.append(
-                    rec.sample_uniform_global(sim, sample_region_scale=1.0)
+        # test the Receptacle instances
+        num_test_samples = 10
+        for receptacle in test_receptacles:
+            # check for contents and correct type parsing
+            if receptacle.name == "receptacle_aabb_chair_test":
+                assert type(receptacle) is hab_receptacle.AABBReceptacle
+            elif receptacle.name == "receptacle_mesh_chair_test":
```
[the remark this answers is not in the document]
Comment (JHurricane96) [reply test/test_rearrange_task.py]: Updated this based on the diff.

**Review by ykarmesh — APPROVED** [review]

**Pull request merged**